PBI-01 Create Assignment

johaneschristian opened the pull request:
Modelled and create functions to support the creation of an assignment that can be used in an assessment event. At the current implementation, the assessment tools can only be created by an assessor, from which the associated company data will be obtained.

johaneschristian committed:
[RED] create stub and model for assessment creation process

johaneschristian committed:
[RED] create tests for assignment creation

johaneschristian committed:
[GREEN] implement functionalities to create assignment

johaneschristian committed:
[REFACTOR] add require_POST decorator on serve_create_assignment

Added require_POST decorator to restrict request type to be only of
type POST, according to the suggestions made by SonarCloud.

whenwe177 commented on assessment/views.py:
```diff
@@ -0,0 +1,17 @@
+@api_view(['POST'])
+@permission_classes([IsAuthenticated])
+def serve_create_assignment(request):
+    request_data = json.loads(request.body.decode('utf-8'))
```
If possible Jo, can you specify the request format for creating assignments?

johaneschristian replied:
Sure, I will include it as the function's documentation using docstring.

johaneschristian committed:
[REFACTOR] specify request format for serve_create_assignment

johaneschristian committed:
[RED] create stub and test for sanitize_file_format

The sanitize_file_format function will be used to clean user's input on
the expected_file_format field. To ensure uniformity, the file_format
will be stored without the leading . (dot).

johaneschristian committed:
[GREEN] implement sanitize_file_format function
